rllib/evaluation/collectors/simple_list_collector.py

```diff
@@ -191,7 +215,101 @@ def add_action_reward_next_obs(self, values: Dict[str, TensorType]) -> None:
                     sub_list.append(flattened[i])
         self.agent_steps += 1
 
-    def build(self, view_requirements: ViewRequirementsDict) -> SampleBatch:
+    def build_for_inference(self) -> SampleBatch:
+        """During inference we will a samplebatch with a batch size of 1.
+        This data will only include the data for the last recorded timestep.
+
+        Returns:
+            A SampleBatch with a batch size of 1.
+        """
+
+        batch_data = {}
+        np_data = {}
+        for view_col, view_req in self.view_requirements.items():
+            # Create the batch of data from the different buffers.
+            data_col = view_req.data_col or view_col
+
+            # if this view is not for inference, skip it.
+            if not view_req.used_for_compute_actions:
+                continue
+            else:
+                if np.any(view_req.shift_arr > 0):
```

Comment:
you can actually take this check out of else. one fewer level of indentation.

Reply:
sg